src/goto-instrument/contracts/instrument_spec_assigns.cpp

```diff
@@ -513,3 +682,117 @@ void instrument_spec_assignst::invalidate_heap_and_spec_aliases(
   for(const auto &pair : from_heap_alloc)
     invalidate_car(pair.second, freed_car, dest);
 }
+
+/// Returns true iff an `ASSIGN lhs := rhs` instruction must be instrumented.
+bool instrument_spec_assignst::must_check_assign(
+  const goto_programt::const_targett &target,
+  skip_function_paramst skip_function_params,
+  const optionalt<cfg_infot> cfg_info_opt)
+{
+  if(
+    const auto &symbol_expr =
+      expr_try_dynamic_cast<symbol_exprt>(target->assign_lhs()))
+  {
+    if(
+      skip_function_params == skip_function_paramst::No &&
+      ns.lookup(symbol_expr->get_identifier()).is_parameter)
+      return true;
+
```

Comment:
Nit pick: braces around the body of an if statement with a multi-line decision, please.